src/query/sql/src/planner/optimizer/property/selectivity.rs

```diff
@@ -292,14 +289,8 @@ impl<'a> SelectivityEstimator<'a> {
             if !self.updated_column_indexes.contains(index) {
                 let new_ndv = (column_stat.ndv * selectivity).ceil();
                 column_stat.ndv = new_ndv;
-                if let Some(histogram) = &mut column_stat.histogram {
-                    if new_ndv as u64 <= 2 {
-                        column_stat.histogram = None;
-                    } else {
-                        for bucket in histogram.buckets.iter_mut() {
-                            bucket.update(selectivity);
-                        }
-                    }
+                if selectivity < 0.8 {
+                    column_stat.histogram = None;
```

Comment:
This part is related to columns that **are not used** in a predicate. So there is a better to maintain their histogram, right? @Dousir9

Reply:
I think it is better to keep these histograms, as these columns may be used in joins.

Reply:
These columns, do not directly by pruning buckets to generate a histogram, but need to use the series of errors I mentioned above to regenerate the histogram, the follow-up may be used, bringing the error, of course, directly with the ndv is also an error. There may be a better way, we need to investigate the business of mainstream databases, such as orcale, db2 way, and then decide the next step.

Reply:
Yeah, we can temporarily keep the previous method, which is: we assume that the columns are not correlated and then scale down each bucket of the histogram by selectivity proportionally.

Reply:
I'd like to set them to None first, so that it's easier to determine the root cause of the problem when doing job/tpcds benchmark later. This prevents the problem from being let go because of `-- -> +`.

Reply:
In the previous implementation, this histogram will also be discarded after it is used in join (used once or multiple times, depending on whether this column is used in join equi conditions), can you give an example of `-- -> +`.